app/frontend/controllers/app_controller.js

```diff
@@ -502,50 +548,36 @@ class AppController {
       fullScreenButton.setAttribute('title', i18n.t('menu.exit-fullscreen'));
       fullScreenButton.firstElementChild.classList.add('fa-compress');
       fullScreenButton.firstElementChild.classList.remove('fa-expand');
+      this.verse_list_popup.disableNewTabButton();
+      wheelnavController.bindEvents();
+      this.hideMenu();
+
+      app_controller.tag_assignment_menu.moveTagAssignmentList("POPUP");
     } else {
       fullScreenButton.setAttribute('title', i18n.t('menu.fullscreen'));
       fullScreenButton.firstElementChild.classList.add('fa-expand');
       fullScreenButton.firstElementChild.classList.remove('fa-compress');
-    }
-  }
-  
-  getLineBreak() {
-    if (process.platform === 'win32') {
-      return "\r\n";
-    } else {
-      return "\n";
-    }
-  }
+      this.verse_list_popup.enableNewTabButton();
+      wheelnavController.unbindAndClose();
+      this.showMenu();
 
-  async copySelectedVersesToClipboard() {
-    const { clipboard } = require('electron');
-    var bibleTranslationId = app_controller.tab_controller.getTab().getBibleTranslationId();
-    var separator = await i18nHelper.getReferenceSeparator(bibleTranslationId);
-    
-    var selectedVerseBoxes = app_controller.verse_selection.selected_verse_box_elements;
-    
-    var selectedText = "";
-    var multipleVerses = selectedVerseBoxes.length > 1;
-
-    for (var i = 0; i < selectedVerseBoxes.length; i++) {
-      var currentVerseBox = $(selectedVerseBoxes[i]);
-      var verseReferenceContent = currentVerseBox.find('.verse-reference-content').text();
-      var currentVerseNr = verseReferenceContent.split(separator)[1];
-      
-      var currentText = currentVerseBox.find('.verse-text').clone();
-      currentText.find('.sword-markup').remove();
-
-      if (multipleVerses) {
-        selectedText += currentVerseNr + " ";
+      if (!app_controller.optionsMenu._tagListOption.isChecked) {
+        app_controller.tag_assignment_menu.moveTagAssignmentList("PREVIOUS");
       }
-
-      selectedText += currentText.text().trim() + " ";
     }
+  }
+
+  async copySelectedVerseTextToClipboard() {
+    var selectedVerseText = await this.verse_selection.getSelectedVerseText();
+    this.getPlatform().copyTextToClipboard(selectedVerseText);
+  }
```

Comment:
Maybe this can be a part of `verse_selection` component?

Reply:
Ok